src/Swagger/Serializer/ApiGatewayNormalizer.php

```diff
@@ -94,18 +96,18 @@ public function normalize($object, $format = null, array $context = [])
                     unset($data['definitions'][$definition]['properties'][$property]['readOnly']);
                 }
                 if (isset($propertyOptions['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $propertyOptions['$ref'])) {
-                    $data['definitions'][$definition]['properties'][$property]['$ref'] = str_replace(['-', '_'], '', $propertyOptions['$ref']);
+                    $data['definitions'][$definition]['properties'][$property]['$ref'] = str_replace(['-', ':', '_'], '', $propertyOptions['$ref']);
                 }
                 if (isset($propertyOptions['items']['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $propertyOptions['items']['$ref'])) {
-                    $data['definitions'][$definition]['properties'][$property]['items']['$ref'] = str_replace(['-', '_'], '', $propertyOptions['items']['$ref']);
+                    $data['definitions'][$definition]['properties'][$property]['items']['$ref'] = str_replace(['-', ':', '_'], '', $propertyOptions['items']['$ref']);
                 }
             }
         }
 
         // $data['definitions'] is an instance of \ArrayObject
         foreach (array_keys($data['definitions']->getArrayCopy()) as $definition) {
             if (!preg_match('/^[A-z]+$/', (string) $definition)) {
-                $data['definitions'][str_replace(['-', '_'], '', (string) $definition)] = $data['definitions'][$definition];
+                $data['definitions'][str_replace(['-', ':', '_'], '', (string) $definition)] = $data['definitions'][$definition];
```

Comment:
In this case, it's safe to use `preg_replace` to get rid of all non-alphanumeric characters, yes?

Reply:
On first sight, probably yes.

Reply:
This is the model name, right? So API Gateway does not allow any non-alphanumeric character here.

Reply:
`$data['definitions'][preg_replace('/[^\da-z]/i', '', (string) $definition)] = $data['definitions'][$definition];` this works.

Reply:
`a-z` is wrong. It should be `A-z`

Reply:
Oh, `/i` flag, my bad.